### PR TITLE
Task 14 representing relations

### PR DESCRIPTION
**Pull request by VasiaBirchak**

### CODE REVIEW

File: myblog/blog/api/serializers.py [hunk 1]
```diff
@@ -8,3 +8,23 @@ class PostSerializer(serializers.ModelSerializer):
...
+        model = BlogPost
+        fields = ('title', 'id')
+
+
+class CommentGETPatchSerializer(serializers.ModelSerializer):
+    blogpost = PostSummarySerializer()
+
+    class Meta:
+        model = Comment
+        fields = '__all__'
```
Comment (ejariza-evowill) [inline myblog/blog/api/serializers.py:24]: don't use `__all__`, instead put the name of all fields that you're going to use in this serializer.

Remember: _explicit is better than implicit._

File: myblog/blog/api/serializers.py [hunk 2]
```diff
@@ -8,3 +8,23 @@ class PostSerializer(serializers.ModelSerializer):
...
+
+
+class CommentPostSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Comment
+        fields = '__all__'
```
Comment (ejariza-evowill) [inline myblog/blog/api/serializers.py:30]: Same here

File: myblog/blog/tests/test_comment.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+from django.test import TestCase
+import pytest
+from django.contrib.auth.models import User
+from blog.models import BlogPost, Comment
+from django.core.exceptions import ValidationError
+
+
+@pytest.mark.django_db
+class TestCommentModel(TestCase):
```
Comment (ejariza-evowill) [inline myblog/blog/tests/test_comment.py:9]: Once again, stick to use `pytest`. `TestCase` is from `unittest`

File: myblog/blog/tests/test_comment.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+    def setUp(self):
+        super().setUp()
+        self.user = User.objects.create_user(username='testuser', password='1234')
+        self.post = BlogPost.objects.create(title='Test Post', body='Test Body', author=self.user)
```
Comment (ejariza-evowill) [inline myblog/blog/tests/test_comment.py:14]: These should be fixtures as well.

File: myblog/blog/views.py [hunk 5]
```diff
@@ -48,15 +47,27 @@ def blog_index(request):
...
+class CommentListCreateAPIView(generics.ListCreateAPIView):
+    queryset = Comment.objects.all()
+
+    def get_serializer_class(self):
+        if self.request.method == 'POST':
+            return CommentPostSerializer
+        return CommentGETPatchSerializer
+
+
+class CommentRetrieveUpdateDestroyAPIView(generics.RetrieveUpdateDestroyAPIView):
```
Comment (ejariza-evowill) [inline myblog/blog/views.py:67]: These two classes are identical, Why don't making a single one that inherits from `ModelViewSet`?

**Review by ejariza-evowill — CHANGES_REQUESTED** [review]

File: myblog/blog/urls.py [hunk 6]
```diff
@@ -1,19 +1,15 @@
...
 from . import views
-from .views import CommentListCreateAPIView, PostViewSet
+from .views import PostViewSet
 from rest_framework.routers import DefaultRouter
-from .views import CommentRetrieveUpdateDestroyAPIView
 
 router = DefaultRouter()
 router.register(r'api/post', PostViewSet, basename='post')
-
+router.register(r'api/comments', views.CommentViewSet, basename='comment')
```
Comment (ejariza-evowill) [inline myblog/blog/urls.py:8]: Keep consistency here, whether you import both viewsets from views or only import views, in other words, adapt one of this styles:

```python
router.register(r'api/post', views.PostViewSet, basename='post')
# or
router.register(r'api/comments', CommentViewSet, basename='comment')
```

**Review by ejariza-evowill — CHANGES_REQUESTED** [review]

File: myblog/blog/urls.py (hunk 6 above)
Comment (VasiaBirchak) [reply myblog/blog/urls.py:8]: ok. I understand

**Review by ejariza-evowill — APPROVED** [review]

**Comment by ejariza-evowill** [comment]
Before proceeding with new tasks, fix conflicts here and merge,

**Pull request merged**